RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,200 @@
+
+```{r sf_files, live = TRUE}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` 
+package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character by `/`, the fourth item is the sample 
+name.
+This is exactly what `stringr::word` allows us to do: split up the file paths
+by `/` and extract the sample names.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, 4, sep = "/")
+sample_names
+```
+
+`tximeta` needs a data.frame with at least these two columns: 
+- a `files` column  with the file paths to the quant.sf files
+- a `names` column with the sample names
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names, 
+                      stringsAsFactors = FALSE)
+```
+
+### Set up metadata
+
+And now we'll read in the sample metadata.
+
+```{r sample_meta_df, live = TRUE}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file)
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`. 
+
+```{r join-sample_meta_df}
+coldata <- coldata %>%
+  dplyr::inner_join(sample_meta_df, by = c("names" = "srr_accession"))
+
+coldata
+```
+
+## Set up our tximeta object
+
+Using our `coldata` data frame we set up, we can run the `tximeta()` which will 
+extract species specific information.
+
+```{r read_tsv, live = TRUE}
+txi_meta <- tximeta(coldata)
+```
+
+*There are extra steps needed if you are using any species other than human or mouse.
+See our other species examples in `tximeta_for_other_species.Rmd`. 
+
+### Summarize to gene
+
+We'll summarize to the gene level using the `summarizeToGene()` function.
+
+```{r make-txi}
+# Summarize to the gene-level
+gene_summarized <- summarizeToGene(txi_meta) 
+```
+
+We can use the `class` function to see what type of object `gene_summarized` is.
+
+```{r class-txi, live = TRUE}
+# Check what type of object `gene_summarized` is
+class(gene_summarized)
+```
+
+This tells us that `gene_summarized` is an object called a [`SummarizedExperiment`](https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/SummarizedExperiment-class) which can be handled by functions from the package of the same name. 
+We more specifically have a `RangedSummarizedExperiment` which is a more specific type of `SummarizedExperiment`.
+
+`SummarizedExperiment` objects have this general structure:  
+
+![SummarizedExperiment](diagrams/SummarizeExperiment-structure.png)
+
+This figure is from this handy vignette about [`SummarizedExperiment` objects](https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html).
+
+The `assay` slot in `SummarizedExperiment`s hold the experiment data, a gene x sample matrix. 
+In this case, it has our gene-level abundance information. 
```

Comment:
I think it is worth calling out the row data and column data at this point as well. People will be happy to see that `rowData(gene_summarized)` has more than just ENSG IDs; it has symbols too!